### ISSUE TITLE
Discuss: question mark operator

### ISSUE THREAD

**Issue by IanManske** [nushell/nushell#12614]
### Related problem

This is a discussion issue around how, if desired, a question mark operator should work in the nu language. There are two main motivations behind this:
1. to allow more ergonomic error handling as opposed to `try`.
2. to provide a unified way for handling errors.

Regarding `2.`, some commands are strict and can be expected to error (e.g., the `first` and `into` family of commands). This is why `get` has the `-i` option. But rather than adding a `-i` flag to each of these commands, a question mark operator would provide a standard method for ignoring errors from any command.

### Describe the solution you'd like

One proposal is that the question mark operator would serve as a shorthand to `try`. If a value is not an error, it is returned. Otherwise, the question mark operator returns `null`.

So, the following two would be equivalent:
```nushell
[] | try { first }
[] | first? 
```

And the same for these two examples:
```nushell
try { [a] | into int | first }
([a] | into int | first)?
```

But since this adds no functionality over `try` and is simply syntactic sugar, another possible solution is to have `?` return a boolean. The use case here is to detect whether or not an internal or external command failed:
```nushell
# `?` with an error returns `false`,
# otherwise `true` is returned.

# can be read as: if this did not succeed
if not (mkdir "test")? {
  # print an error or do some other logic
}

# can be read as: if this succeeded
if (mkdir "test")? {
  # ...
}
```

This kind of check would not be possible with the first solution that returns `null`, since `mkdir` also returns `null`.

Then, this can be extended to test the exit code of an external command:
```nushell
if (^false)? {
  # zero exit code
} else {
  # non-zero exit code 
}
```

**Comment by fdncred** [comment]
Yes, this is a bit confusing since we already use it for `get -i` scenarios to make cell paths optional. You don't say that until the last sentence, so I was like, "maybe he means just extending the ? operator", LOL.

Back to the issue, I'm not really a fan of turning try/catch into some abbreviated symbol because it starts to make nushell have symbology like bash that is even more confusing imo like 2>&1.

I maybe would support adding a finally though, which is a separate subject.

**Comment by KAAtheWiseGit** [comment]
Nushell allows question marks in commands, so, theoretically, one could define an alias:

```Nushell
~> alias first? = try { first }
~> [] | first? | describe
nothing
```

But in general, I think that such an operator isn't a good fit for a shell.  A somewhat more ergonomic alternative might be postfix try: sort of like `complete` works, but for the built-in commands.
